job_creator.php

```diff
@@ -93,6 +93,14 @@ public function getInstallerVersion(): string
         // fallback to use the next-minor or latest-minor version of installer
         $installerVersions = array_keys(INSTALLER_TO_PHP_VERSIONS);
         $installerVersions = array_filter($installerVersions, fn($version) => substr($version, 0, 1) === $cmsMajor);
+
+        // Force CMS 6 branches to only return 6.x-dev instead of 6.0.x-dev
+        // as the 6.0 branches won't exist until the first beta release
+        // remove this code when the 6.0 branches exist
+        if ($cmsMajor == '6') {
+            return '6.x-dev';
+        }
```

Comment:
Is there a way we can dynamically check if the `6.0` branch exists? Or rather if the `"{$cmsMajor}.0"` branch exists? Then we could keep this for future major releases as well.

Reply:
Not easily.

We probably (though possibly issues since in context of github actions) file_get_contentes('https://api.github.com/repos/silverstripe/silverstripe-framework/branches') and take it from there

But it'll a pain with unit tests cos we can't have them doing live pings. We could maybe mock the response, but it's really not worth the effort here

When we create the 6.0 branches for the beta we'll get feedback really quick that it's using the wrong version of installer and we'll go looking for the reason and we'll find this pretty quick

Note - even if we did using file_get_contents() the beta will still result in a sea of red builds on the 6.0 branches cos all the things branched before framework branch 6.0 exists will still try to use installer 6.x-dev

I'd say just leave as is

Reply:
We should be able to get the list of branches directly from `git` (we do that elsewhere in CI I'm pretty sure), no need to hit an API to get a list of branches. Regardless of how we get the branch list, we could do it as an earlier step and store it in a property. Then in tests just set that property to whatever we want it to be for the test we're running.

> When we create the 6.0 branches for the beta we'll get feedback really quick that it's using the wrong version of installer and we'll go looking for the reason and we'll find this pretty quick

It would be better to not have that happen in the first place. But this is less about 6.0 for me and more for 7, 8, 9 etc etc. I don't want us doing this same work over and over again.
I lost the push for future proofing our CI when we did 5, which is why this PR exists in the first place.

Reply:
We don't want the list of local git branches, we only care about whether or not the relevant branch of `silverstripe/installer` exists, so we need API

I've updated this to see if a minor branch of silverstripe/installer e.g. `6.0`.  If it doesn't then it will fallback to using the major branch e.g. `6`